lib/Advisor.js

```diff
@@ -111,67 +144,162 @@ function Advisor(options) {
         ]).spread(_generateAdvice);
     };
 
+    /**
+     * Convenience method that can be used to find the least utilized
+     * instance on demand.
+     *
+     * @return {Promise<Status>}
+     */
+    this.getLeastUtilizedTarget = () => {
+        return this.getAllStatuses().then((statuses) => {
+            statuses = statuses.filter((status) => {
+                return _isTargetHealthy(status);
+            });
+
+            if (statuses.length === 0) {
+                // we can always return ourself (not great but better than
+                // throwing error?)
+                return _selfStatus;
+            } else {
+                statuses.sort(statusValueSorter);
+                return statuses[0];
+            }
+        });
+    };
+
     /* private functions */
     function _publishSelfStatus(){
         return _getSelfStatus()
             .then((statusParameters) => {
                 let status = new Status(statusParameters);
-                _selfStatus = status;
                 status.validate();
+
+                // put address of this instance into status
+                status.address = _selfAddress;
+
+                // put name of this instance into status
+                status.name = _selfName;
+
+                // record our current timestamp in status
+                status.timestamp = Date.now();
+
                 return _dbInterface.publishSelfStatus(status);
             });
     }
 
     function _getPeerStatuses(){
-        return _dbInterface.getPeerStatuses()
-            .tap(function(peerStatuses) { _peerStatuses = peerStatuses; });
+        return _dbInterface.getPeerStatuses().tap(function(peerStatuses) {
+            _peerStatuses = peerStatuses;
+        });
+    }
+
+    function _isTargetHealthy(target) {
+        if (target.maxCapacity === 0) {
+            return false;
+        }
+
+        if (_healthyThreshold) {
+            let timeDiff = Math.abs(Date.now() - target.timestamp);
+            if (timeDiff > _healthyThreshold) {
+                // skip this instance because it appears unhealthy
+                // according to the constraints we were given
+                return false;
+            }
+        }
+
+        return true;
     }
 
     function _generateAdvice(selfStatus, peerStatuses) {
-        let allStatuses = [selfStatus].concat(peerStatuses);
-        if (allStatuses.length < 2) {
+        if (!peerStatuses.length) {
```

Comment:
there is no advice for when there is 1 or 0 instances in the cluster. this was an optimization for when there is only one instance.